keylime/cloud_verifier_tornado.py

```diff
@@ -249,48 +365,62 @@ def put(self):
         try:
             rest_params = common.get_restful_params(self.request.uri)
             if rest_params is None:
-                common.echo_json_response(self, 405, "Not Implemented: Use /agents/ interface")
+                common.echo_json_response(
+                    self, 405, "Not Implemented: Use /agents/ interface")
                 return
 
             if "agents" not in rest_params:
                 common.echo_json_response(self, 400, "uri not supported")
-                logger.warning('PUT returning 400 response. uri not supported: ' + self.request.path)
+                logger.warning(
+                    'PUT returning 400 response. uri not supported: ' + self.request.path)
                 return
 
             agent_id = rest_params["agents"]
+
             if agent_id is None:
                 common.echo_json_response(self, 400, "uri not supported")
                 logger.warning("PUT returning 400 response. uri not supported")
+            try:
+                agent = session.query(VerfierMain).filter_by(agent_id=agent_id)
+            except SQLAlchemyError as e:
+                logger.error(f'SQLAlchemy Error: {e}')
 
-            agent = self.db.get_agent(agent_id)
-
-            if agent is not None:
+            if agent is None:
                 common.echo_json_response(self, 404, "agent id not found")
-                logger.info('PUT returning 404 response. agent id: ' + agent_id + ' not found.')
+                logger.info(
+                    'PUT returning 404 response. agent id: ' + agent_id + ' not found.')
 
             if "reactivate" in rest_params:
-                agent['operational_state']=cloud_verifier_common.CloudAgent_Operational_State.START
-                asyncio.ensure_future(self.process_agent(agent, cloud_verifier_common.CloudAgent_Operational_State.GET_QUOTE))
+                agent['operational_state'] = cloud_verifier_common.CloudAgent_Operational_State.START
```

Comment:
Does this part of the code work?  When an agent fails it executes this line and throws the exception: 
`TypeError: 'Query' object does not support item assignment`

It seems like this should be `agent.operational_state = ....`

Reply:
I am not that my work station at the moment, but agree this should be `agent.operational_state` - are you able to check that, if not I can in the morning?

Reply:
It seems to fix the issue on my box at least (Ubuntu)

Reply:
Made a ToDo to check this and push a commit tomorrow!

Reply:
Ok, make sure it works on your side as well.  I also haven't tested IMA functionality yet.